src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/FindSmallIndelRegions.java

```diff
@@ -40,44 +62,152 @@ public boolean requiresReads()
+                    for ( int idx = 0; idx != cdfLen; ++idx ) {
+                        writer.write(String.format("%.3f\n", cdf.getFraction(idx)));
                     }
-                    finder.checkHistograms(evList);
-                    return evList.iterator();
-                })
-                .collect();
-
-        System.out.println();
-        final List<SAMSequenceRecord> seqs = header.getSequenceDictionary().getSequences();
-        for ( final BreakpointEvidence evidence : evidenceList ) {
-            final SVInterval interval = evidence.getLocation();
-            final String contigName = seqs.get(interval.getContig()).getSequenceName();
-            System.out.println(contigName + "\t" + (interval.getStart() + 1) + "\t" + interval.getEnd());
+                } catch ( final IOException ioe ) {
+                    throw new UserException("Unable to write sample CDF " + fileName, ioe);
+                }
+            }
+        }
+        allReads
+            .mapPartitions(readItr -> {
+                final List<String> statList = new ArrayList<>(EVIDENCE_SIZE_GUESS);
+                final Finder finder = new Finder(broadcastReadMetadata.value(), filter,
+                                                    broadcastTrueIntervals.value(), cdfDirname);
+                while ( readItr.hasNext() ) {
+                    finder.testReadAndGatherStatus(readItr.next(), statList);
+                }
+                finder.getStatus(statList);
+                return statList.iterator();
+            })
+            .saveAsTextFile(outputFile);
+    }
+
+    @DefaultSerializer(SVTypeLen.Serializer.class)
+    public final static class SVTypeLen {
+        private SvType.TYPES type;
+        private int len;
+
+        public SVTypeLen( final String svType, final int len ) {
+            this.type = SvType.TYPES.valueOf(svType);
+            this.len = len;
+        }
+
+        public SVTypeLen( final Kryo kryo, final Input input ) {
+            type = SvType.TYPES.values()[input.readInt()];
+            len = input.readInt();
+        }
+
+        public void serialize( final Kryo kryo, final Output output ) {
+            output.writeInt(type.ordinal());
+            output.writeInt(len);
+        }
+
+        public SvType.TYPES getType() { return type; }
+        public int getLen() { return len; }
+
+        public static final class Serializer extends com.esotericsoftware.kryo.Serializer<SVTypeLen> {
+            @Override
+            public void write( final Kryo kryo, final Output output, final SVTypeLen interval ) {
+                interval.serialize(kryo, output);
+            }
+
+            @Override
+            public SVTypeLen read( final Kryo kryo, final Input input, final Class<SVTypeLen> klass ) {
+                return new SVTypeLen(kryo, input);
+            }
         }
     }
 
+    private static SVIntervalTree<SVTypeLen> readTruthFile( final String truthFile,
```

Comment:
Currently in master, the VCF output is guaranteed to have a SVLEN field. But when it comes the BND and IMPRECISE records, no such guarantee can be made (BND records shouldn't even have a `SVLEN` IMO).